src/system/SystemLayerImplSelect.cpp

```diff
@@ -526,12 +639,47 @@ void LayerImplSelect::HandleEvents()
 }
 
 #if CHIP_SYSTEM_CONFIG_USE_DISPATCH
+
 void LayerImplSelect::HandleTimerComplete(TimerList::Node * timer)
 {
     mTimerList.Remove(timer);
     mTimerPool.Invoke(timer);
 }
-#endif // CHIP_SYSTEM_CONFIG_USE_DISPATCH
+
+#elif CHIP_SYSTEM_CONFIG_USE_LIBEV
+
+void LayerImplSelect::HandleLibEvTimer(EV_P_ struct ev_timer * t, int revents)
+{
+    TimerList::Node * timer = static_cast<TimerList::Node *>(t->data);
+    VerifyOrDie(timer);
+    LayerImplSelect * layerP = dynamic_cast<LayerImplSelect *>(timer->mCallback.mSystemLayer);
+    VerifyOrDie(layerP);
+    layerP->mTimerList.Remove(timer);
+    layerP->mTimerPool.Invoke(timer);
```

Comment:
Can you please remind me again what ensures that things queued on the platform manager event queue and these timer callbacks don't run in parallel in this libuv setup?

Reply:
This is now a bit embarassing: nothing ensures that YET - in the PR as it was posted originally :-(

But you did ask about that back in September (in #22043), and I did take care of it through a change in `GenericPlatformManagerImpl_POSIX`, but forgot to update the PR (done now):

When compiling with `CHIP_SYSTEM_CONFIG_USE_LIBEV`, no separate thread is started for dispatching platform events, but these are posted via ScheduleWork which finally ends up as a libev timer, and thus gets executed sequentially on the main thread.

Reply:
This is now included in the current PR (rebased and cleaned up according to discussion)